tools/integration_tests/test_storage.py

```diff
@@ -266,11 +266,9 @@ def test_get_collection_only(self):
         # Sets the maximum number of ids that will be returned
         self.retry_delete(self.root + '/storage/xxx_col2')
 
-        bsos = []
         for i in range(10):
             bso = {'id': str(i).zfill(2), 'payload': 'x', 'sortindex': i}
-            bsos.append(bso)
-        self.retry_post_json(self.root + '/storage/xxx_col2', bsos)
+            self.retry_post_json(self.root + '/storage/xxx_col2', [bso])
```

Comment:
I still don't understand why this change was needed, I'll run these tests and take a look.

Were you seeing it fail solely on the Spanner emulator?

Reply:
We just chatted about this, but posting this for posterity:

My understanding is that the tests starting on L298 expect the BSOs to be sorted based on timestamp, but since we insert all of the records at once, all of the BSOs have the same timestamp. This leads to intermittent test failures -- the `modified` timestamps are all equal, so when sorting on that field, the order of the BSOs returned by Spanner is nondeterministic. I've only seen the failure on the emulator, but I'm guessing that's due to implementation differences between the emulator and the hosted database. It just so happens that the hosted version of Spanner has been returning the BSOs in a consistent order between the `GET` calls on lines 299 and 303, but I don't think there's any guarantee that that will continue to be the case

Reply:
Sorry I just got back around to looking at this.

Though the changes to the tests are subtle and small I think it would be safe to leave them as is (the assertEquals changes are probably even more significant than the bulk post change).

I think we can solve this another way by adding something like a "consistent_sort_order" flag to `SpannerDb` that enables the commented out issue559 section. This adds both a secondary `bso_id` stablilization sort (that spanner does implicitly since it's the primary key) and defaults no sort specified to act like sort=newest.

We might want to look into enabling this again on production Spanner but IIRC it caused a slow down.

Interestingly enough the Python version has this exact same flag for kind of the opposite reason -- the extra sorting block slowed down older Mysql's but not modern ones.

Reply:
We chatted via slack about this; I'll go ahead and key the stabilization sort off of `spanner_emulator_host` 👍🏻